flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  inputs = {
+    nixpkgs.url = "github:serokell/nixpkgs";
+    haskell-nix.url = "github:input-output-hk/haskell.nix";
+    hackage = {
+      url = "github:input-output-hk/hackage.nix";
+      flake = false;
+    };
+    stackage = {
+      url = "github:input-output-hk/stackage.nix";
+      flake = false;
+    };
+    common-infra.url = "github:serokell/common-infra";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage, stackage, common-infra }:
+    let
+      packagesFor = system: rec {
+        pkgs = nixpkgs.legacyPackages.${system}.extend
+          (nixpkgs.lib.foldl nixpkgs.lib.composeExtensions (_: _: { }) [
+            haskell-nix.overlay
+          ]);
+        backend = pkgs.callPackage ./backend { };
+      };
+    in {
+      packages = builtins.mapAttrs (system: _:
+        with packagesFor system; {
+          backend-lib = backend.library;
+          backend = backend.server // {
+            meta.modulePath = [ "services" "edna" "backend" ];
+          };
+        }) nixpkgs.legacyPackages;
+
+      ciSystems = [ "x86_64-linux" ];
```

Comment:
Btw, it seems very unlikely that we will have more than one `ciSystem` in any foreseeable future, so this file can be simplified, we don't need to iterate over systems and pass them to functions, we can have a single constant `ciSystem`.

Reply:
This isn't used anywhere. `mkPipelineFile` only generates steps for `x86_64-linux`.